### PR TITLE
Updating Atlassian connector and Jira addon to SDK v3

### PR DESCRIPTION
**Pull request by yavorg**
Additional fixes made:
- The connector was collecting an Atlassian domain/host name property that was unused and misleading, removed
- Upgraded the logic in the addon to cache the access token and its expiration, instead of getting a new one on every request

### CODE REVIEW

File: addons/atlassian-jira/uninstall.js [hunk 1]
```diff
@@ -3,16 +3,22 @@ This is the uninstallation logic of the Lifecycle Manager.
 */
 
 const Sdk = require('@fusebit/add-on-sdk');
-const Superagent = require('superagent');
 
 module.exports = async (ctx) => {
```
Comment (bennbollay) [inline addons/atlassian-jira/uninstall.js]: Consider try/catch around each failable call - few things should cause a deletion to fail in a way that each resource doesn't get at least a best-effort deletion.

File: connectors/atlassian/install.js [hunk 2]
```diff
@@ -7,14 +7,17 @@ const getTemplateFile = (fileName) => Fs.readFileSync(__dirname + `/template/${f
 const Sdk = require('@fusebit/add-on-sdk');
 
 module.exports = async (ctx) => {
+  if (!ctx.caller.permissions) {
```
Comment (bennbollay) [inline connectors/atlassian/install.js]: Side note for future: it seems like there's a consistent pattern where things like install, uninstall, etc require permissions but a few others do not.  Might be worth codifying that in the manager.

Comment (bennbollay) [reply connectors/atlassian/install.js]: Just to be clear: not proposing any change in this PR.  Mark this comment as resolved once you've read it :)

Comment (yavorg) [reply connectors/atlassian/install.js]: Makes sense to make this a part of the manager, there is definitely a very clear pattern. I've filed a ticket to track

File: addons/atlassian-jira/uninstall.js (hunk 1 above)
Comment (yavorg) [reply addons/atlassian-jira/uninstall.js]: We decided to fail here so we have visibility at least for now into if/how things are failing. Also, this file is uniform across all addons at the moment. I'm not opposed to switching to something more forgiving there, but not as part of this PR.

**Pull request merged**